websocket/src/websocket.cpp

```diff
@@ -242,13 +242,19 @@ static void CloseConnection(WebsocketConnection* conn)
     // we want it to send this message in the polling
     if (conn->m_State == STATE_CONNECTED) {
 #if defined(HAVE_WSLAY)
+        // close the connection and immediately transition to the DISCONNECTED
+        // state
+        SetState(conn, STATE_DISCONNECTED);
         WSL_Close(conn->m_Ctx);
 #else
+        // start disconnecting by closing the WebSocket through the JS API
+        // we transition to the DISCONNECTED state when we receive the
+        // Emscripten callback that the connection has closed
+        SetState(conn, STATE_DISCONNECTING);
```

Comment:
(very) small nitpick: It's good practice to change state _after_ the current state is finished. (Imagine a threaded application querying state)